src/pydisagg/ihme/preprocess.py

```diff
@@ -0,0 +1,417 @@
+from typing import Tuple
+
+import numpy as np
+import pandas as pd
+from typing import Dict, List
+
+from .age_var import (
+    age_id_map,
+    match_pats,
+    match_pops,
+)
+
+
+def validate_data(df, pop_df, age_id_map=age_id_map):
+    """
+    Validates the input dataframe by checking for NaN values, invalid sex_id, invalid location_ids, and invalid year_ids.
```

Comment:
As tedious as it sounds, I will separate this function into a set of sub-function, for example
```
def validate_nonan(...): ...
def validate_sex_id(...): ...
def validate_location_id(...): ...
def validate_year_id(...): ...
```
Will make it much easier to test and read